modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1210,25 +1302,13 @@ public final void setCellFactory(Callback<VirtualFlow<T>, T> value) {
             // the currentIndex, then we will adjust the position to be 1.
             // Otherwise, our goal is to leave the index of the cell at the
             // top consistent, with the same translation etc.
-            if (position == 0 || position == 1) {
-                // Update the item count
-//                setItemCount(cellCount);
-            } else if (currentIndex >= cellCount) {
+            if (position != 0 && position != 1 && (currentIndex >= cellCount)) {
```

Comment:
Comparing a double for equality or inequality is not the best coding practice. Anyway, I see this pattern throughout this file. We can live with it for now.

Reply:
The comparison was already in the code. I agree though it is brittle in general, although it can probably be proved that in this case, the value of position will never be e.g. 1+\delta with \delta as small as possible.
But it would be good practice for a follow-up issue to remove the equality checks on doubles in this file (and others).

Reply:
I agree that this shouldn't be changed as part of this fix, since it is preexisting. Filing one or more follow-on issues to address this more globally seems good.